src/ServantElm.hs

```diff
@@ -166,6 +238,118 @@ mkRequest request =
 
     expect =
       case request ^. reqReturnType of
-        Just _ ->
-          "Http.expectWhatever toMsg"
+        Just elmTypeExpr ->
+          "Http.expectJson" <+> toMsg <+> (typeRefDecoder . definitionToRef) elmTypeExpr
         Nothing -> error "mkHttpRequest: no reqReturnType?"
+
+    body =
+      case request ^. reqBody of
+        Just elmTypeExpr ->
+          "Http.jsonBody" <+> parens ((typeRefEncoder . definitionToRef) elmTypeExpr <+> bodyValue)
+        Nothing ->
+          "Http.emptyBody"
+
+typeRefDecoder :: TypeRef -> Doc ann
```

Comment:
This is taken from `elm-street`, right? You should add a comment, so that we know what we should ask `elm-street` to expose from their api eventually.

Reply:
okay, done